dynamic_search.py

```
@@ -0,0 +1,163 @@
+        comic_ids = [row[0] for row in self.cursor.fetchall()]
+        if not comic_ids:
+            return None
+
+        placeholders = ",".join("?" for _ in comic_ids)
+
+        self.cursor.execute(f"""
+            SELECT comics.*, publishers.name AS publisher_name
+            FROM comics
+            LEFT JOIN publishers ON comics.publisher_id = publishers.id
+            WHERE comics.id IN ({placeholders})
+        """, comic_ids)
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)